### ISSUE TITLE
Do not warn about names with digits

### ISSUE THREAD

**Issue by Pzixel** [rust-lang/rust#47060]
Currently I have following code
```rust
const LOG4RS_File : &str = "log4rs.yaml";
```
And I get following warning:

```
warning: constant `LOG4RS_File` should have an upper case name such as `LOG4_RS_FILE`
 --> build.rs:7:1
  |
7 | const LOG4RS_File : &str = "log4rs.yaml";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: #[warn(non_upper_case_globals)] on by default
```

But library name is `LOG4RS` is defenitly uppercased `log4rs`, but compiler thinks different.

I think there should be no warning. Numbers in the middle of the name are ok: log4net, write2file and so on.

____

rustc 1.22.1 (05e2e1c41 2017-11-22)
binary: rustc
commit-hash: 05e2e1c41414e8fc73d0f267ea8dab1a3eeeaa99
commit-date: 2017-11-22
host: x86_64-pc-windows-gnu
release: 1.22.1
LLVM version: 4.0

**Comment by retep998** [comment]
It is not complaining about the number, nor the `LOG4RS` part. It is actually complaining about the `File` part which should instead be `FILE`.
